File: app/src/main/java/com/example/todonotes/DetailActivity.java
package com.example.todonotes;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

public class DetailActivity extends AppCompatActivity {
    String TAG = "DetailActivity";
    TextView textViewTitle, textViewDescription;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);
        bindViews();
        setupIntent();
    }

    private void setupIntent() {
        Intent intent = getIntent();
        Log.d(TAG, intent.getStringExtra(AppConstant.TITLE));
        String title = intent.getStringExtra(AppConstant.TITLE);
        String description = intent.getStringExtra(AppConstant.DESCRIPTION);
        textViewTitle.setText(title);
        textViewDescription.setText(description);
    }

    private void bindViews() {
        textViewTitle = findViewById(R.id.textViewTitle);
        textViewDescription = findViewById(R.id.textViewDescription);
    }
}

File: app/src/main/java/com/example/todonotes/DetailActivity.kt
package com.example.todonotes

import android.os.Bundle
import android.util.Log
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class DetailActivity: AppCompatActivity() {
    var TAG = "DetailActivity"
    lateinit var textViewTitle: TextView
    lateinit var textViewDescription:TextView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)
        bindViews()
        setupIntent()
    }

    private fun bindViews() {
        textViewTitle = findViewById(R.id.textViewTitle)
        textViewDescription = findViewById(R.id.textViewDescription)
    }

    private fun setupIntent() {
        val intent = intent  // getIntent()
        Log.d(TAG, intent.getStringExtra(AppConstant.TITLE)!!)
        val title = intent.getStringExtra(AppConstant.TITLE)
        val description = intent.getStringExtra(AppConstant.DESCRIPTION)
        // setText()
        textViewTitle.text = title
        textViewDescription.text = description
    }
}